pulser-core/pulser/channels/channels.py

```diff
@@ -366,48 +368,62 @@ def validate_pulse(self, pulse: Pulse) -> None:
             )
 
     def modulate(
-        self, input_samples: np.ndarray, keep_ends: bool = False
+        self,
+        input_samples: np.ndarray,
+        keep_ends: bool = False,
+        eom: bool = False,
     ) -> np.ndarray:
         """Modulates the input according to the channel's modulation bandwidth.
 
         Args:
             input_samples: The samples to modulate.
             keep_ends: Assume the end values of the samples were kept
                 constant (i.e. there is no ramp from zero on the ends).
+            eom: Whether to calculate the modulation using the EOM
+                bandwidth.
 
         Returns:
             The modulated output signal.
         """
-        if not self.mod_bandwidth:
+        if eom:
+            if not self.supports_eom():
+                raise TypeError(f"The channel {self} does not have an EOM.")
+            eom_config = cast(BaseEOM, self.eom_config)
+            mod_bandwidth = eom_config.mod_bandwidth
+            rise_time = eom_config.rise_time
+
+        elif not self.mod_bandwidth:
             warnings.warn(
                 f"No modulation bandwidth defined for channel '{self}',"
                 " 'Channel.modulate()' returns the 'input_samples' unchanged.",
                 stacklevel=2,
             )
             return input_samples
+        else:
+            mod_bandwidth = self.mod_bandwidth
+            rise_time = self.rise_time
 
         # The cutoff frequency (fc) and the modulation transfer function
         # are defined in https://tinyurl.com/bdeumc8k
-        fc = self.mod_bandwidth * 1e-3 / np.sqrt(np.log(2))
+        fc = mod_bandwidth * 1e-3 / np.sqrt(np.log(2))
         if keep_ends:
-            samples = np.pad(input_samples, 2 * self.rise_time, mode="edge")
+            samples = np.pad(input_samples, 2 * rise_time, mode="edge")
         else:
-            samples = np.pad(input_samples, self.rise_time)
+            samples = np.pad(input_samples, rise_time)
         freqs = fftfreq(samples.size)
         modulation = np.exp(-(freqs**2) / fc**2)
         mod_samples = ifft(fft(samples) * modulation).real
```

Comment:
This operation could be potentially expensive depending on the sample size (FFT --> modulation --> IFFT). Is it executed every time before sequence execution?

Reply:
I'm not sure what you mean by "sequence execution", but I can tell you that it is not called in any time-sensitive part. Potential delays are confined to taking a bit longer to draw a sequence, pre-process it for simulation (ie when creating the time-dependent Hamiltonian) and potentially when adding a very long pulse.

Reply:
Thanks, understood.